### ISSUE TITLE
No logits_warper applied in guided_decoding method

### ISSUE THREAD

**Issue by yw-ucsb** [voidism/Lookback-Lens#7]
Hi,

Thanks for sharing this impressive work! This repo has been very helpful in my recent research. However, when I check the guided decoding method, I found that no logits warper is passed to this method. My understanding of the paper is that the guided decoding process is to call multiple rounds of the "sample" method to generate candidate chunks, so that the sampling process should ideally stay consistent with the default sample method. I checked that the default sampling process from huggingface is as below:
```
......
elif generation_mode == GenerationMode.SAMPLE:
            # 11. prepare logits warper
            logits_warper = self._get_logits_warper(generation_config)

            # 12. expand input_ids with `num_return_sequences` additional sequences per batch
            input_ids, model_kwargs = self._expand_inputs_for_generation(
                input_ids=input_ids,
                expand_size=generation_config.num_return_sequences,
                is_encoder_decoder=self.config.is_encoder_decoder,
                **model_kwargs,
            )

            # 13. run sample
            return self.sample(
                input_ids,
                logits_processor=logits_processor,
                logits_warper=logits_warper,
                stopping_criteria=stopping_criteria,
                pad_token_id=generation_config.pad_token_id,
                eos_token_id=generation_config.eos_token_id,
                output_scores=generation_config.output_scores,
                return_dict_in_generate=generation_config.return_dict_in_generate,
                synced_gpus=synced_gpus,
                streamer=streamer,
                **model_kwargs,
            )
......
```
The guided decoding method does not perform step 11 above: (thought I found an input argument in its implementation)

``` 
......
if guiding_classifier is not None:
            # 11. classifier guided decoding
            return self.classifier_guided_decoding(
                input_ids,
                logits_processor=logits_processor,
                stopping_criteria=stopping_criteria,
                pad_token_id=generation_config.pad_token_id,
                eos_token_id=generation_config.eos_token_id,
                output_scores=generation_config.output_scores,
                return_dict_in_generate=generation_config.return_dict_in_generate,
                synced_gpus=synced_gpus,
                extra_prompt_length=extra_prompt_length,
                guiding_classifier=guiding_classifier,
                chunk_size=chunk_size,
                num_candidates=num_candidates,
                conversion_matrix=conversion_matrix,
                feat_layer=feat_layer,
                **model_kwargs,
            )
......
```

I'm wondering if there is something I miss and I'd appreciate if you can help me with this question. Thanks again!

**Comment by voidism** [comment]
Hi,

Sorry for the late reply! And sorry that I haven't check the issues of this repo for a long time. 
I just checked the code and confirm that you're right, the logit wrapper is not initialized in the function of `classifier_guided_decoding`, but the logits_processor and stopping_criteria are correctly initialized so I didn't found this problem before.

The issue potentially caused by this can be the temperature and top-p/top-k is not set.
We originally intended to use temperature 0.9 and top-p sampling with p = 0.95, now it's set to default: temp 1.0 and top-p 1.0. This will include slightly more randomness when generating chunk candidates in our classifier guided decoding process, but may not be crucial to change our results.

To keep the results consistent with our original experiment, I think I have to keep the original code (so temp 1.0 and top-p 1.0 are set). However, thanks so much for being so careful reading our code and pointing this error out!
